app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,32 @@
+class CommentsController < ApplicationController
+  before_action :find_article, only: %i[create]
+  before_action :find_comment, only: %i[destroy]
+
+  http_basic_authenticate_with name: 'user', password: 'secret', only: :destroy
+
+  def create
+    @comment = @article.comments.create(comments_params)
+    redirect_to article_path(@article)
+  end
+
+  def destroy
+    if @comment.article_id == params[:article_id].to_i
```

Comment:
Please refactor this to make the code cleaner.

Reply:
Done !